enterprise/cmd/frontend/internal/authz/resolvers/resolver.go

```diff
@@ -544,6 +552,91 @@ func (r *permissionsInfoResolver) Unrestricted() bool {
 	return r.unrestricted
 }
 
+var permissionsInfoRepositoryConnectionMaxPageSize = 100
+
+var permissionsInfoRepositoryConnectionOptions = &graphqlutil.ConnectionResolverOptions{
+	OrderBy:     database.OrderBy{{Field: "repo.name"}},
+	Ascending:   true,
+	MaxPageSize: &permissionsInfoRepositoryConnectionMaxPageSize,
+}
+
+func (r *permissionsInfoResolver) Repositories(ctx context.Context, args graphqlbackend.PermissionsInfoRepositoriesArgs) (*graphqlutil.ConnectionResolver[graphqlbackend.PermissionsInfoRepositoryResolver], error) {
+	if r.userID == 0 {
+		return nil, nil
+	}
+
+	query := ""
+	if args.Query != nil {
+		query = *args.Query
+	}
+
+	connectionStore := &permissionsInfoRepositoriesStore{
+		userID: r.userID,
+		db:     r.db,
+		ossDB:  r.ossDB,
+		query:  query,
+	}
+
+	return graphqlutil.NewConnectionResolver[graphqlbackend.PermissionsInfoRepositoryResolver](connectionStore, &args.ConnectionResolverArgs, permissionsInfoRepositoryConnectionOptions)
+}
+
+type permissionsInfoRepositoriesStore struct {
+	userID int32
+	db     edb.EnterpriseDB
+	ossDB  database.DB
+	query  string
+}
+
+func (s *permissionsInfoRepositoriesStore) MarshalCursor(node graphqlbackend.PermissionsInfoRepositoryResolver, _ database.OrderBy) (*string, error) {
+	cursor := string(node.Repository().Name())
+
+	return &cursor, nil
+}
+
+func (s *permissionsInfoRepositoriesStore) UnmarshalCursor(cursor string, _ database.OrderBy) (*string, error) {
+	return &cursor, nil
+}
+
+func (s *permissionsInfoRepositoriesStore) ComputeTotal(ctx context.Context) (*int32, error) {
+	total := int32(20)
+	return &total, nil
+}
+
+func (s *permissionsInfoRepositoriesStore) ComputeNodes(ctx context.Context, args *database.PaginationArgs) ([]graphqlbackend.PermissionsInfoRepositoryResolver, error) {
+	permissions, err := s.db.Perms().ListUserPermissions(ctx, s.userID, &edb.ListUserPermissionsArgs{Query: s.query, PaginationArgs: args})
+	if err != nil {
+		return nil, err
+	}
+
+	var permissionResolvers []graphqlbackend.PermissionsInfoRepositoryResolver
```

Comment:
We know the size upfront, so might initialize it as such:
```suggestion
	permissionResolvers := make([]graphqlbackend.PermissionsInfoRepositoryResolver, 0, len(permissions))
```